### PR TITLE
Use unique table names in distributed tests

### PR DESCRIPTION
**Pull request by kokosing**
Use unique table names in distributed tests

### CODE REVIEW

**Comment by kokosing** [comment]
This is still not complete list of tables used in tests.... but I hope it will be enough

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 1]
```diff
@@ -195,63 +195,55 @@ public void testCreateTable()
     @Test
     public void testCreateTableAsSelect()
     {
-        assertUpdate("CREATE TABLE IF NOT EXISTS test_ctas AS SELECT name, regionkey FROM nation", "SELECT count(*) FROM nation");
-        assertTableColumnNames("test_ctas", "name", "regionkey");
-        assertUpdate("DROP TABLE test_ctas");
+        String tableName = "test_ctas" + randomTableSuffix();
+        assertUpdate("CREATE TABLE IF NOT EXISTS " + tableName + " AS SELECT name, regionkey FROM nation", "SELECT count(*) FROM nation");
+        assertTableColumnNames("" + tableName, "name", "regionkey");
```
Comment (sopel39) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:200]: `"" + tableName` doesn't make sense

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 2]
```diff
@@ -195,63 +195,55 @@ public void testCreateTable()
     @Test
     public void testCreateTableAsSelect()
     {
-        assertUpdate("CREATE TABLE IF NOT EXISTS test_ctas AS SELECT name, regionkey FROM nation", "SELECT count(*) FROM nation");
-        assertTableColumnNames("test_ctas", "name", "regionkey");
-        assertUpdate("DROP TABLE test_ctas");
+        String tableName = "test_ctas" + randomTableSuffix();
+        assertUpdate("CREATE TABLE IF NOT EXISTS " + tableName + " AS SELECT name, regionkey FROM nation", "SELECT count(*) FROM nation");
```
Comment (sopel39) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:199]: consider using `String.format` instead

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 3]
```diff
@@ -527,171 +519,176 @@ public void testInsertWithCoercion()
     @Test
     public void testInsertUnicode()
     {
-        assertUpdate("DROP TABLE IF EXISTS test_insert_unicode");
+        String tableName = "test_insert_unicode_" + randomTableSuffix();
+
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
```
Comment (sopel39) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:524]: I don't think it's needed anymore

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 4]
```diff
@@ -801,7 +814,7 @@ public void testViewMetadata()
         MaterializedResult expected = resultBuilder(getSession(), actual.getTypes())
                 .row("customer", "BASE TABLE")
                 .row("lineitem", "BASE TABLE")
-                .row("meta_test_view", "VIEW")
+                .row("" + viewName, "VIEW")
```
Comment (sopel39) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:817]: `"" + viewName` doesn't make sense

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 5]
```diff
@@ -829,13 +842,13 @@ public void testViewMetadata()
                 getSession().getSchema().get()));
 
         expected = resultBuilder(getSession(), actual.getTypes())
-                .row("meta_test_view", formatSqlText(query))
+                .row("" + viewName, formatSqlText(query))
```
Comment (sopel39) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:845]: ditto

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java (hunk 2 above)
Comment (findepi) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:199]: when fmt-ing SQLs in tests, using + is often more readable than %s (especially when you substitude very few things in long text)

**Comment by electrum** [comment]
Why do we want to do this? Are there connectors that use a shared schema when running tests concurrently?

You could verify this manually by adding a hack in `ConnectorMetadata` to make sure any created tables match a 32-char hex regex.

**Comment by kokosing** [comment]
> Are there connectors that use a shared schema when running tests concurrently?

Yes. Actually, they are running concurrently multiple distributed queries suites with substantially different setup.

> You could verify this manually by adding a hack in ConnectorMetadata to make sure any created tables match a 32-char hex regex.

Good idea. WIll use it next time I hit this. I don't want to spend too much time on it ATM. 
Also, still it won't be future proof.

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java (hunk 2 above)
Comment (kokosing) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:199]: Also, I refactored this code with some search-and-replace methods. Using `format` would be much harder to make it automatically.

**Pull request merged**